service/jmx/src/main/java/org/jolokia/service/jmx/handler/WriteHandler.java

```diff
@@ -119,7 +120,7 @@ private Object setAttribute(JolokiaWriteRequest request, MBeanServerConnection s
      * @throws IllegalAccessException if access to MBean fails
      * @throws InvocationTargetException reflection error when setting an object's attribute
      */
-    private Object[] getValues(String pType, Object pCurrentValue, JolokiaWriteRequest pRequest)
+    private WriteRequestValues getValues(String pType, Object pCurrentValue, JolokiaWriteRequest pRequest)
```

Comment:
I see you point to avoid an untyped Object[] array as return value. However, I wonder whether we could make `WriteRequestValue` a inner class here and avoid it in the `Serializer` interface.